article-store/events/utils.py

```diff
@@ -0,0 +1,122 @@
+from contextlib import contextmanager
+from functools import partial, wraps
+import json
+from time import gmtime, strftime
+from typing import (
+    Any,
+    Callable,
+    ContextManager,
+    Dict
+)
+import uuid
+
+import pika
+from pika.exceptions import ConnectionClosed
+from pika.adapters.blocking_connection import BlockingChannel
+
+from .conf import (
+    ARTICLES_EXCHANGE,
+    BROKER_PARAMS,
+    DEFAULT_QUEUES
+)
+
+
+DELIVERY_MODE_PERSISTENT = 2
+
+
+def create_message(msg_type: str, identifier: str, data: Dict) -> Dict:
+    """Create a message `dict` based on a standard schema.
+
+    :return: Dict
+    """
+    return {
+        "eventId": str(uuid.uuid1()),
+        "happenedAt": strftime("%Y-%m-%dT%H:%M:%S+00:00", gmtime()),
+        "aggregate": {
+            "service": "article-store",
+            "name": "article-version",
+            "identifier": identifier,
+        },
+        "type": msg_type,
+        "data": data or {}
+    }
+
+
+@contextmanager
+def get_channel() -> ContextManager[BlockingChannel]:
+    """Handles the creation and clean up of a connection,
+    giving the caller a connection channel to use.
+    :return: class: `BlockingChannel`
+    """
+    try:
+        connection = pika.BlockingConnection(parameters=BROKER_PARAMS)
+        yield connection.channel()
+        connection.close()
+    except ConnectionClosed as err:
+        print(err)
+        raise
+
+
+def setup_exchanges_and_queues(func) -> Callable[..., None]:
+    """Setup required queues and exchanges on target broker.
+
+    If they exist already they will be skipped.
+
+    :return:
+    """
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        with get_channel() as channel:
+            for queue_name in DEFAULT_QUEUES:
+                channel.queue_declare(queue=queue_name, durable=True)
+
+                for exchange in DEFAULT_QUEUES[queue_name]:
+                    channel.exchange_declare(exchange=exchange, exchange_type='fanout', durable=True)
```

Comment:
exchanges (only `ARTICLES_EXCHANGE`) should be declared even if there are no queues to be bound to them; this code can be pulled up to the start of the `with get_channel()` block